Sources/HummingbirdAuth/Sessions/UserSessionRepository.swift

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2021-2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Hummingbird
+import Logging
+
+/// Context object supplied when requesting user
+public struct UserRepositoryContext {
+    public let logger: Logger
+
+    public init(logger: Logger) {
```

Comment:
Does this init need to be public? I'd prefer making it internal so we can expand it with non-optional properties